broker/broker.go

```diff
@@ -308,8 +318,21 @@ func (b *CdnServiceBroker) checkDomain(domain, orgGUID string) error {
 }
 
 func (b *CdnServiceBroker) getHeaders(options Options) []string {
+	headers := make([]string, len(options.Headers))
+	copy(headers, options.Headers)
+	// Preserve original 'Host' header if forwarding to CloudFoundry's routing table.
 	if options.Origin == b.settings.DefaultOrigin {
-		return []string{"Host"}
+		// Do not append 'Host' if forwarding all headers.
+		if len(headers) == 1 && headers[0] == "*" {
```

Comment:
While working on this I realised that the logic around `*` headers belongs outside of the `options.Origin == b.settings.DefaultOrigin` check. A `*` header should always be provided alone and isn't specific to the `DefaultOrigin`, whereas appending `Host` is.